component/component.go

```diff
@@ -219,7 +219,9 @@ type Artifact interface {
 	Labels() map[string]string
 }
 
-type Deployment interface{}
+type Deployment interface{
+	URL() string
```

Comment:
We should make this optional to: (1) prevent breaking existing plugins and (2) prevent an empty string return for most plugins since this is not the most common.

Reply:
Solved, I'm now using a `DeploymentWithUrl` as you suggested in the Waypoint PR